runtime/src/in_mem_accounts_index.rs

```diff
@@ -396,7 +398,7 @@ impl<T: IndexValue> InMemAccountsIndex<T> {
                             false
                         }
                     }
-                    None => false, // not in cache or on disk
+                    None => true, // not in cache or on disk, but slot list is 'empty' and entry is not in index, so return true
```

Comment:
Note this is a change in behavior. The correct thing to do here is return true, since we are now being more specific about what the return value from this function means.